modules/cluster-autoscaler/main.tf

```diff
@@ -9,12 +17,13 @@ locals {
 }
 
 module "iam" {
-  source = "github.com/terraform-aws-modules/terraform-aws-iam//modules/iam-assumable-role-with-oidc?ref=v2.10.0"
+  source = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  # source = "github.com/terraform-aws-modules/terraform-aws-iam//modules/iam-assumable-role-with-oidc?ref=v2.10.0"
 
   create_role                   = var.enable
   role_name                     = "${var.cluster_name}-cluster-autoscaler-irsa"
-  provider_url                  = var.oidc_provider_issuer_url
-  oidc_fully_qualified_subjects = ["system:serviceaccount:${kubernetes_namespace.cluster_autoscaler.0.metadata.0.name}:aws-cluster-autoscaler"]
+  provider_url                  = local.provider_url
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${local.namespace}:aws-cluster-autoscaler"]
```

Comment:
Should probably not hard-code the service account name either. There's a value for the chart you can set iirc.